tests/unit/rules/javascript/test.unsupported_browser_api.js

```diff
@@ -121,4 +121,67 @@ describe('unsupported browser APIs', () => {
     const { linterMessages } = await runJsScanner(jsScanner);
     expect(linterMessages.length).toEqual(0);
   });
+
+  describe('Experiement APIs', () => {
+    it.each([
+      {
+        title: 'allows known privileged APIs when the extension is privileged',
+        privileged: true,
+        expectedMessages: 0,
+      },
+      {
+        title: 'emits an error when the extension is not privileged',
+        privileged: false,
+        expectedMessages: 1,
+      },
+    ])('$title', async ({ privileged, expectedMessages }) => {
+      const code = 'browser.some.experiment.foo()';
+      const jsScanner = new JavaScriptScanner(code, 'api.js', {
+        addonMetadata: {
+          experimentApiPaths: ['some.experiment'],
+        },
+        privileged,
+      });
+
+      const { linterMessages } = await runJsScanner(jsScanner);
+      expect(linterMessages.length).toEqual(expectedMessages);
```

Comment:
could we also assert that we got the expected reported linting issue along with checking we got the expected number of linting issues reported?

Reply:
I think so